python/cudf/cudf/tests/conftest.py

```diff
@@ -1,10 +1,49 @@
 import pathlib
 
+import cupy as cp
+import numpy as np
 import pytest
 
 import rmm  # noqa: F401
 
+from cudf.testing._utils import assert_eq
+
 
 @pytest.fixture(scope="session")
 def datadir():
     return pathlib.Path(__file__).parent / "data"
+
+
+@pytest.fixture(
+    params=[None, 42, np.random.RandomState, cp.random.RandomState]
+)
+def random_state_tuple(request):
```

Comment:
Nice, this looks good! It's a little convoluted, but I think that's fine given what it's trying to achieve. It's pretty specialized for the sampling tests though, so could we document that here? In the event that someone wants something similar in the future it would be good for them to understand what they can't break when modifying this fixture.

Reply:
Done.